apps/core/scenes/bookingDetail/sectorDetails/segments/SegmentContainer.js

```diff
@@ -0,0 +1,78 @@
+// @flow
+
+import * as React from 'react';
+import {
+  Touchable,
+  Icon,
+  Text,
+  StyleSheet,
+} from '@kiwicom/universal-components';
+import { defaultTokens } from '@kiwicom/orbit-design-tokens';
+import { Animated } from 'react-native';
+
+type Props = {||};
+
+type State = {|
+  +expandSegments: boolean,
+|};
+
+export default class SegmentContainer extends React.Component<Props, State> {
+  constructor() {
+    super();
+
+    this.state = {
+      expandSegments: false,
+    };
+
+    this.spinValue = new Animated.Value(0);
+  }
+
+  spinValue: Animated.Value;
+
+  toggleExpanded = () => {
+    Animated.timing(this.spinValue, {
```

Comment:
When I tap on arrow quickly multiple times it can end up in wrong position.
Maybe we can disable button/toggle/state updates during animation.

<img width="349" alt="screenshot 2019-02-20 at 15 25 35" src="https://user-images.githubusercontent.com/2660330/53098514-16418c00-3524-11e9-9059-85ff43a16ce2.png">

Otherwise I love it.

Reply:
Yeah, I will have a look at that 👍